panther_manager/src/docking_manager_node.cpp

```diff
@@ -0,0 +1,125 @@
+
+#include <panther_manager/behavior_tree_manager.hpp>
+#include <panther_manager/behavior_tree_utils.hpp>
+
+namespace panther_manager
+{
+
+DockingManagerNode::DockingManagerNode(
+  const std::string & node_name, const rclcpp::NodeOptions & options)
+: Node(node_name, options)
+{
+  RCLCPP_INFO(this->get_logger(), "Constructing node.");
+
+  DeclareParameters();
+  const std::map<std::string, std::any> docking_initial_bb = {};
+  const int bt_server_port = this->get_parameter("bt_server_port").as_int();
+
+  docking_tree_manager_ = std::make_unique<BehaviorTreeManager>(
+    "Docking", docking_initial_bb, bt_server_port);
+
+  RCLCPP_INFO(this->get_logger(), "Node constructed successfully.");
+}
+
+void DockingManagerNode::Initialize()
+{
+  RCLCPP_INFO(this->get_logger(), "Initializing.");
+
+  RegisterBehaviorTree();
+  docking_tree_manager_->Initialize(factory_);
+
+  using namespace std::placeholders;
+
+  const float timer_freq = this->get_parameter("timer_frequency").as_double();
+  const auto timer_period_ms =
+    std::chrono::milliseconds(static_cast<unsigned>(1.0f / timer_freq * 1000));
+
+  docking_tree_timer_ = this->create_wall_timer(
+    timer_period_ms, std::bind(&DockingManagerNode::TimerCB, this));
+
+  RCLCPP_INFO(this->get_logger(), "Initialized successfully.");
+}
+
+void DockingManagerNode::DeclareParameters()
+{
+  const auto panther_manager_pkg_path =
+    ament_index_cpp::get_package_share_directory("panther_manager");
+  const std::string default_bt_project_path = panther_manager_pkg_path +
+                                              "/behavior_trees/DockingBT.btproj";
+  const std::vector<std::string> default_plugin_libs = {};
+
+  this->declare_parameter<std::string>("bt_project_path", default_bt_project_path);
+  this->declare_parameter<std::vector<std::string>>("plugin_libs", default_plugin_libs);
+  this->declare_parameter<std::vector<std::string>>("ros_plugin_libs", default_plugin_libs);
+  this->declare_parameter<double>("ros_communication_timeout.availability", 1.0);
+  this->declare_parameter<double>("ros_communication_timeout.response", 1.0);
+
+  this->declare_parameter<float>("timer_frequency", 20.0);
+  this->declare_parameter<int>("bt_server_port", 4444);
+}
+
+void DockingManagerNode::RegisterBehaviorTree()
+{
+  const auto bt_project_path = this->get_parameter("bt_project_path").as_string();
+
+  const auto plugin_libs = this->get_parameter("plugin_libs").as_string_array();
+  const auto ros_plugin_libs = this->get_parameter("ros_plugin_libs").as_string_array();
+
+  const auto service_availability_timeout =
+    this->get_parameter("ros_communication_timeout.availability").as_double();
+  const auto service_response_timeout =
+    this->get_parameter("ros_communication_timeout.response").as_double();
+
+  BT::RosNodeParams params;
+  params.nh = this->shared_from_this();
+  params.wait_for_server_timeout =
+    std::chrono::milliseconds(static_cast<int>(service_availability_timeout * 1000));
```

Comment:
Can we try like this?
```suggestion
  params.wait_for_server_timeout = std::chrono::duration<double>(service_availability_timeout);
```

Reply:
Expected type is `std::chrono::milliseconds`.

Reply:
okay...  Then we can cast it. 
```suggestion
 auto timeout_s = std::chrono::duration<double>(service_availability_timeout);
  params.wait_for_server_timeout = std::chrono::duration_cast<std::chrono::milliseconds>(timeout_s);
```